src/components/Dropdown/Dropdown.js

```diff
@@ -35,6 +41,35 @@ export class Dropdown extends React.Component {
     this.toggle();
   };
 
+  onTextChange = (e) => {
+    const value = e.target.value;
+    let updatedState = {
+      ...this.state,
+      isOpen: true,
+      selection: value,
+      suggestions: []
+    };
+    if (value.length > 0) {
+      const regex = new RegExp(`^${value}`, 'i');
+      const filteredSuggestions = [...this.state.options]
+        .sort()
+        .filter((v) => regex.test(v));
+      updatedState.suggestions = filteredSuggestions;
+      this.setState({ ...updatedState });
+    }
+    if (!value.length) {
+      updatedState.suggestions = this.state.options;
+      this.setState({ ...updatedState });
+    }
+  };
```

Comment:
onTextChange = ({target: {value}}) => {
    const updatedState = {
      isOpen: true,
      selection: value,
    };
    if (value.length) {
      const regex = new RegExp(`^${value}`, 'i');
      const filteredSuggestions = this.state.options
        .filter((v) => regex.test(v))
        .sort();
      this.setState({ ...updatedState, suggestions: filteredSuggestions });
    }else {
      this.setState({ ...updatedState });
    }
  };